File: wms-lite-user-web/src/features/master/customer/customer.ts
/**
 * 고객사 (Customer) 도메인 관련 타입 정의
 * 백엔드 wms-lite-server (com.wms.wms_lite.domain.master.customer) DTO 스펙 대응
 */

export interface CustomerResponse {
  id: number;
  code: string;
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CustomerSummaryResponse {
  id: number;
  code: string;
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
}

export interface CustomerSearchRequest {
  keyword?: string;
  page?: number;
  size?: number;
}

export interface CustomerCreateRequest {
  code: string;
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
}

export interface CustomerUpdateRequest {
  name: string;
  businessNo?: string;
  ceoName?: string;
  phone?: string;
  email?: string;
}

File: wms-lite/wms-lite/src/main/java/com/wms/wms_lite/domain/master/customer/entity/Customer.java
package com.wms.wms_lite.domain.master.customer.entity;

import com.wms.wms_lite.domain.master.customer.enums.CustomerStatus;
import com.wms.wms_lite.global.entity.SoftDeleteEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "customers")
public class Customer extends SoftDeleteEntity {

    @Column(nullable = false, unique = true, length = 50)
    private String code;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(length = 30)
    private String businessNo;

    @Column(length = 100)
    private String ceoName;

    @Column(length = 30)
    private String phone;

    @Column(length = 150)
    private String email;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private CustomerStatus status = CustomerStatus.ACTIVE;
}
